internal/clone/registry.go

```diff
@@ -0,0 +1,161 @@
+package clone
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/regstate"
+	"github.com/Microsoft/hcsshim/internal/uvm"
+)
+
+const (
+	configRoot = "croot"
+	configKey  = "ckey"
+)
+
+type persistedUVMConfig struct {
+	// actual information related to template / clone
+	RawData []byte
+	// metadata field used to determine if this config is already saved.
+	Stored bool
+}
+
+// When encoding interfaces gob requires us to register the struct types that we will be
+// using under those interfaces. This registration needs to happen on both sides i.e the
+// side which encodes the data and the side which decodes the data.
+// Go init function: https://golang.org/doc/effective_go.html#init
```

Comment:
I assume what this means by "sides" is the shim related to the template and the shim related to the clone, right? Might be worth changing this a bit :)